src/main/java/com/google/sps/data/Contest.java

```diff
@@ -26,41 +27,53 @@
 public abstract class Contest {
   public abstract String getName();
 
-  public abstract HashMap<Long, Candidate> getCandidates();
+  public abstract HashSet<String> getCandidates();
 
   public abstract String getDescription();
 
   public static Builder builder() {
     return new AutoValue_Contest.Builder();
   }
 
-  // creates a new Contest object by extracting the properties from "obj"
-  public static Contest fromJSONObject(JSONObject obj) throws JSONException {
+  @AutoValue.Builder
+  public abstract static class Builder {
+    public abstract Builder setName(String name);
+
+    public abstract Builder setCandidates(HashSet<String> candidates);
+
+    public abstract Builder setDescription(String description);
+
+    public abstract Contest build();
+  }
+
+  // creates a new Contest object by extracting the properties from "contestData"
```

Comment:
You should really note any side effects that take place from functions (in this case, the contests are inserted into the datastore).